### PR TITLE
Update python and cibuildwheel versions

### LINKED ISSUE #645: Python 3.12 support
I am wondering if there is a chance for a Python 3.12 version.

### PR DESCRIPTION
**Pull request by ATM-Jahid**
This should resolve #645.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Base Python (debian) image
-FROM python:3.11
+FROM python:3.12
```
Comment (jameskermode) [inline docker/Dockerfile]: Is this strictly required? If not I suggest we leave it out of this PR.

Comment (ATM-Jahid) [reply docker/Dockerfile]: Reverted the change in the dockerfile.

**Comment by jameskermode** [comment]
Thanks! I don't think the failures are related to your changes, but I'll try to fix them on separate PR and then ask you to rebase on the updated `main` first so we don't get a mangled release.

**Comment by ATM-Jahid** [comment]
Thank you, @jameskermode.
I am assuming the build errors have something to do with f90wrap.

**Comment by ATM-Jahid** [comment]
I changed the upper bound of f90wrap to use a version that precedes the current `public` commit 694dc9c. QUIP gets built, but it fails on the tests. So, I'm adding an upper bound for numpy as well.

**Comment by jameskermode** [comment]
Thanks, yes, this is likely what's needed. Numpy 2 will break a lot of builds.

**Comment by jameskermode** [comment]
The remaining test failures are due to a recent change in ASE. I'll push a fix for them shortly then we can get this merged.

**Pull request merged**